platform-plugins.md

```diff
@@ -0,0 +1,277 @@
+
+* [SystemSound](https://docs.flutter.io/flutter/services/SystemSound-class.html):
+Provides access to the library of short system specific sounds for common tasks.
+
+* [pathProvider](https://docs.flutter.io/flutter/services/PathProvider-class.html):
+Returns commonly used locations on the filesystem.
+
+See the [services library](https://docs.flutter.io/flutter/services/services-library.html)
+for full details.
+
+**Example: Using the Flutter Platform plugin to read the contents of the Clipboard**
+
+The core Flutter Platform Plugin is available through the [services
+library](https://docs.flutter.io/flutter/services/services-library.html), as
+illustrated in this example:
+
+```dart
+import 'package:flutter/services.dart';
+
+class _MyHomePageState extends State<MyHomePage> {
+ String clipboardContents = 'Unknown';
+
+ getClipboardContents() async {
+   var clipboardData = await Clipboard.getData(Clipboard.kTextPlain);
+   if (clipboardData != null) {
+     setState(() {
+       clipboardContents = clipboardData.text;
+     });
+   }
+ }
+ ...
+}
+```
+
+### Platform Plugins from the Flutter ecosystem
+
+{% include note.html content="Sharing of Platform Plugins is currently not
+supported, but Flutter is expected to support this in the Future." %}
+
+## Create a Platform Plugin {#create}
+
+Flutter uses a flexible system that allows you to call *any* platform-specific
+API whether available in Java-code on Android, or in ObjectiveC- or Swift-code
+on iOS.
+
+Flutter's platform-specific API support does not rely on fragile code
+generation, but rather on a flexible message passing style:
+
+* The Flutter portion of your app sends messages to it's *host*, the iOS or
+ Android portion of your app, over a platform channel.
+
+* The *host* listens on the platform channel, and receives the message. It then
+ calls into any number of platform-specific APIs -- using their native
+ programming language -- and sends back a response to the *client*, the Flutter
+ portion.
+
+### Architectural overview: Platform Channels
+
+Messages are passed between the client (UI) and host (platform) using Platform
+channels as illustrated in this diagram:
+
+![Platform Channels architecture](/images/PlatformChannels.png)
+
+Message and responses are passed asynchronously, to ensure the user interface
+remains responsive.
+
+On the client side, the class
+[`PlatformMessageChannel`](https://docs.flutter.io/flutter/services/PlatformMessageChannel-class.html)
+enables sending general purpose message, and the class
+[`PlatformMethodChannel`](https://docs.flutter.io/flutter/services/PlatformMethodChannel-class.html)
+enables spending specific messages that correspond to method calls. On the
+platform side, the class `FlutterMessageChannel` enables receiving messages
+messages from the client, and responding with a result, and the class
+`FlutterMethodChannel`. These classes allow you to developer a platform plugin
```

Comment:
typo: "developer"

And, choose if we capitalize P for platform. Above you say "Platform channels" and here you say "platform plugin"